plugins/org.python.pydev.parser/src/org/python/pydev/parser/prettyprinterv2/PrettyPrinterVisitorV2.java

```diff
@@ -122,18 +122,36 @@ public Object visitAssign(Assign node) throws Exception {
         java.util.List<ILinePart> recordChanges = null;
         org.python.pydev.shared_core.structure.Tuple<ILinePart, ILinePart> lowerAndHigher = null;
 
-        for (int i = 0; i < node.targets.length; i++) {
-            exprType target = node.targets[i];
-            if (i >= 1) { //more than one assign
+        if (node.targets.length > 0) {
+            for (int i = 0; i < node.targets.length; i++) {
+                exprType target = node.targets[i];
+                if (i >= 1) { //more than one assign
+                    doc.add(lowerAndHigher.o2.getLine(), lowerAndHigher.o2.getBeginCol(),
+                            this.prefs.getAssignPunctuation(), node);
+                }
+                id = this.doc.pushRecordChanges();
+                target.accept(this);
+                recordChanges = this.doc.popRecordChanges(id);
+                lowerAndHigher = doc.getLowerAndHigerFound(recordChanges);
+            }
+            if (node.type != null) {
                 doc.add(lowerAndHigher.o2.getLine(), lowerAndHigher.o2.getBeginCol(),
-                        this.prefs.getAssignPunctuation(), node);
+                        this.prefs.getTypePunctuationColon(), node);
+                id = this.doc.pushRecordChanges();
+                node.type.accept(this);
+                recordChanges = this.doc.popRecordChanges(id);
+                lowerAndHigher = doc.getLowerAndHigerFound(recordChanges);
             }
+            doc.add(lowerAndHigher.o2.getLine(), lowerAndHigher.o2.getBeginCol(), this.prefs.getAssignPunctuation(),
+                    node);
+        } else if (node.type != null) {
             id = this.doc.pushRecordChanges();
-            target.accept(this);
+            node.type.accept(this);
             recordChanges = this.doc.popRecordChanges(id);
             lowerAndHigher = doc.getLowerAndHigerFound(recordChanges);
+            doc.add(lowerAndHigher.o2.getLine(), lowerAndHigher.o2.getBeginCol(),
+                    this.prefs.getTypePunctuationColon(), node);
```

Comment:
It's a bit ouf of the scope, but please in the line below, also check `if(node.value != null)` before doing `node.value.accept(this);`.

Reply:
As I explained in the comment above, when node.targets is empty, the type of variable is stored in node.value... So I tried to handle the cases in the simplest way and keeping the original code